tiledb/api/examples/aggregates.rs

```diff
@@ -0,0 +1,165 @@
+        &tdb,
+        tiledb::array::ArrayType::Dense,
+        domain,
+    )?
+    .add_attribute(attribute_a)?
+    .build()?;
+
+    tiledb::Array::create(&tdb, QUICKSTART_DENSE_ARRAY_URI, schema)
+}
+
+/// Writes data into the array in row-major order from a 1D-array buffer.
+/// After the write, the contents of the array will be:
+/// [[ 1,  2,  3,  4],
+///  [ 5,  6,  7,  8],
+///  [ 9, 10, 11, 12],
+///  [13, 14, 15, 16]]
+fn write_array() -> TileDBResult<()> {
+    let tdb = tiledb::context::Context::new()?;
+
+    let array = tiledb::Array::open(
+        &tdb,
+        QUICKSTART_DENSE_ARRAY_URI,
+        tiledb::array::Mode::Write,
+    )?;
+
+    let data = vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
+
+    let query = tiledb::query::WriteBuilder::new(array)?
+        .layout(tiledb::query::QueryLayout::RowMajor)?
+        .data_typed(QUICKSTART_ATTRIBUTE_NAME, &data)?
+        .build();
+
+    query.submit().map(|_| ())
+}
+
+/// Query back a slice of our array and print the results to stdout.
+/// The slice on "rows" is [1, 2] and on "columns" is [2, 4],
+/// so the returned data should look like:
+/// [[ _,  2,  3,  4],
+///  [ _,  6,  7,  8],
+///  [ _,  _,  _,  _],
+///  [ _,  _,  _,  _]]]
+/// This should print 6, which is the number of elements in the slice.
+fn get_count() -> TileDBResult<()> {
+    let tdb = tiledb::context::Context::new()?;
+
+    let array = tiledb::Array::open(
+        &tdb,
+        QUICKSTART_DENSE_ARRAY_URI,
+        tiledb::array::Mode::Read,
+    )?;
+
+    let mut query = tiledb::query::ReadBuilder::new(array)?
+        .layout(tiledb::query::QueryLayout::RowMajor)?
+        .apply_aggregate::<u64>(AggregateType::Count, None)?
+        .start_subarray()?
+        .add_range("rows", &[1i32, 2])?
+        .add_range("columns", &[2i32, 4])?
+        .finish_subarray()?
+        .build();
+
+    let results = query.execute()?;
+    println!("{}", results);
+
+    Ok(())
+}
+
+fn get_sum() -> TileDBResult<()> {
+    let tdb = tiledb::context::Context::new()?;
+
+    let array = tiledb::Array::open(
+        &tdb,
+        QUICKSTART_DENSE_ARRAY_URI,
+        tiledb::array::Mode::Read,
+    )?;
+
+    let mut query = tiledb::query::ReadBuilder::new(array)?
+        .layout(tiledb::query::QueryLayout::RowMajor)?
+        .apply_aggregate::<i32>(
+            AggregateType::Sum,
+            Some(QUICKSTART_ATTRIBUTE_NAME.to_string()),
+        )?
+        .start_subarray()?
+        .add_range("rows", &[1i32, 2])?
+        .add_range("columns", &[2i32, 4])?
+        .finish_subarray()?
+        .build();
+
+    let results = query.execute()?;
+    println!("{}", results);
+
+    Ok(())
+}
+
+fn main() {
+    if !array_exists() {
+        create_array().expect("Failed to create array");
+    }
+    write_array().expect("Failed to write array");
+    //get_count().expect("Failed to count array");
+    get_sum().expect("Failed to sum array");
+}
```

Comment:
When the aggregate builder supports doing multiple aggregate calls (which the C API does support, right?) then please add another function to this example which demonstrates doing so

Reply:
it doesn't yet :( but I definitely will when I do support that!